lib/SIL/SILProfiler.cpp

```diff
@@ -185,8 +200,11 @@ struct MapRegionCounters : public ASTWalker {
   std::pair<bool, Expr *> walkToExprPre(Expr *E) override {
     if (auto *IE = dyn_cast<IfExpr>(E)) {
       CounterMap[IE->getThenExpr()] = NextCounter++;
-    } else if (isa<AutoClosureExpr>(E) || isa<ClosureExpr>(E)) {
```

Comment:
After type checking, AutoClosureExpr and ClosureExpr are the same semantically, so whatever you do with one should be done with the other.

Reply:
I'll try getting rid of the distinction between autoclosures and closures here and see if anything breaks.

Reply:
Thanks for pointing this out, some nice simplifications fall out of not having special handling for autoclosures.